pkg/proxy/userspace/proxier.go

```diff
@@ -92,14 +94,28 @@ func logTimeout(err error) bool {
 // ProxySocketFunc is a function which constructs a ProxySocket from a protocol, ip, and port
 type ProxySocketFunc func(protocol v1.Protocol, ip net.IP, port int) (ProxySocket, error)
 
+const (
+	serviceEventAdd    int = 1
+	serviceEventUpdate int = 2
+	serviceEventDelete int = 3
```

Comment:
stylistically, you should `type serviceEventType int` or something, and also use `iota`